src/util/simplify_expr_pointer.cpp

```diff
@@ -417,39 +418,67 @@ bool simplify_exprt::simplify_pointer_offset(exprt &expr)
   return true;
 }
 
+static exprt simplify_inequality_address_of_prepare_op(const exprt &op)
```

Comment:
How about just `look_through_typecasts`, then do the postcondition in the caller?

Reply:
Didn't know that was a thing. Will use that instead.

Reply:
That's not a reference to an existing function, rather a suggested name

Reply:
It does exist in `java_pointer_casts.cpp`, but as a static function. Didn't realise you just meant a rename.

Reply:
ha, coincidence. Factoring to a general `util/` helper is probably sensible.